content/post/2025/02-riscv-formal-extension-for-vc-formal.md

```diff
@@ -0,0 +1,298 @@
+
+A structured conversion process would enable **bidirectional compatibility** between open-source and proprietary formal verification tools.  
+
+### SymbiYosys Flow vs. VC Formal Flow  
+
+Comparison of **SymbiYosys Flow** and **Synopsys VC Formal (VCF)** flows:  
+
+| **Flow**                    | **SymbiYosys (SBY) Flow** | **VC Formal (VCF) Flow** |
+|-----------------------------|--------------------------|--------------------------|
+| **Configuration File**       | `.sby` file (SBY-based) | `.tcl` file (Tcl-based) |
+| **Tool Used**               | SymbiYosys + Yosys       | Synopsys VC Formal       |
+| **Solver Backend**          | Boolector, Yices2, etc.,    | Synopsys’ internal solvers |
+| **Counterexample Debugging** | GUI-based + logs  | GUI-based + logs |
+| **Execution Automation**    | Python, Shell & Makefile  | Python, Shell & Makefile |
+|
+
+## Future Scope
+
+1. **Extending to Other Proprietary Tools**  
+   While this extension integrates Synopsys VC Formal with riscv-formal, the approach can be further extended to other commercial formal verification tools. Automating `.sby` file conversion and structured result handling can help streamline workflows across multiple environments.
+
+2. **Maintaining Compatibility with File Format Changes**  
+   This extension relies on the `.sby` file structure from riscv-formal and the `.tcl` scripting format used by VC Formal. If either of these formats evolve, updates will be needed to ensure compatibility. Keeping the automation scripts adaptable to such modifications will be crucial for long-term usability.
+
+By continuously adapting, this extension can remain a valuable tool for formal verification, extending its reach to new tools and maintaining alignment with updates in the riscv-formal ecosystem.
+
+## Results  
+
+We have successfully developed a complete extension that seamlessly integrates the **riscv-formal framework** with **Synopsys VC Formal (VCF)**, ensuring an efficient, automated, and user-friendly verification experience.  
+
+Using this **extended framework**, we formally verified our **Single Cycle RISC-V core (SCRV32I)** and cross-checked results with **SymbiYosys**. The verification confirms that our core adheres to the **RISC-V ISA specifications**, ensuring correctness and compliance.  
+
+### 🔍 Practical Example 
+To demonstrate the workflow, we have included the **SCRV32I processor** as a reference. This provides a **step-by-step guide** to:  
+- Understanding the verification setup.  
+- Running formal verification on SCRV32I.  
+
+Additionally, we provide a dedicated section on:  
+- Configuring the process for custom RISC-V designs.  
+
+This practical approach ensures that users can **replicate adapt, and extend** the verification flow to their own processor designs.  
+
+## Running the Example SCRV32I Processor (For Understanding the Workflow)
```

Comment:
This section and the following one appear to be identical to the readme.  It might be better to provide a summary/abbreviated version for how to configure a core and run riscv-formal with your extension, linking back to the repo for complete/detailed instructions.